File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdatabalancesheetresponse.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsDataBalanceSheetResponse struct {
	Currency                 string                                   `json:"currency"`
	EarliestAvailableMonth   *time.Time                               `json:"earliestAvailableMonth,omitempty"`
	MostRecentAvailableMonth *time.Time                               `json:"mostRecentAvailableMonth,omitempty"`
	Reports                  []CodatDataContractsDatasetsBalanceSheet `json:"reports"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdatabalancesheetresponse.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsBalanceSheet } from "./codatdatacontractsdatasetsbalancesheet";
import { Expose, Transform, Type } from "class-transformer";


export class CodatPublicApiModelsDataBalanceSheetResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "currency" })
  currency: string;

  @SpeakeasyMetadata()
  @Expose({ name: "earliestAvailableMonth" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  earliestAvailableMonth?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "mostRecentAvailableMonth" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  mostRecentAvailableMonth?: Date;

  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsBalanceSheet })
  @Expose({ name: "reports" })
  @Type(() => CodatDataContractsDatasetsBalanceSheet)
  reports: CodatDataContractsDatasetsBalanceSheet[];
}